test/src/fake_converted_query_test.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:fake_cloud_firestore/src/converter.dart';
+import 'package:fake_cloud_firestore/src/fake_converted_query.dart';
+import 'package:fake_cloud_firestore/src/fake_query_with_parent.dart';
+import 'package:fake_cloud_firestore/src/mock_query.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+
+// ignore: subtype_of_sealed_class
+class MockFakeQueryWithParent<T> extends Mock implements MockQuery<T> {}
+
+class MockConverter<T> extends Mock implements Converter<T> {}
+
+void main() {
+  test('FakeQueryWithParentImpl.limit should throw NoSuchMethodError', () {
+    expect(
+      () => FakeQueryWithParentImpl().limit(1),
+      throwsA(isA<NoSuchMethodError>()),
+    );
+  });
+
+  test('FakeConvertedQuery.limit should return normally', () {
+    // arrange
+    final mockFakeQueryWithParent =
+        MockFakeQueryWithParent<Map<String, dynamic>>();
+    final mockConverter = MockConverter<Map<String, dynamic>>();
+
+    // act
+    final query = FakeConvertedQuery<Map<String, dynamic>>(
```

Comment:
As a rule, our unit tests do not refer to the internal implementations. The only reference we do to the project is when instantiating `FakeFirebaseFirestore`. This is to ensure that the project correctly fakes Firestore.

You can copy/paste this test instead:
```dart
import 'fake_cloud_firestore_test.dart';



  test('`limit` clips results', () async {
    final from = (snapshot, _) => Movie()..title = snapshot['title'];
    final to = (Movie movie, _) => {'title': movie.title};

    final firestore = FakeFirebaseFirestore();

    final moviesCollection = firestore
        .collection('movies')
        .withConverter(fromFirestore: from, toFirestore: to);
    await moviesCollection.add(Movie()..title = 'A long time ago');
    await moviesCollection.add(Movie()..title = 'Robot from the future');
    final searchResults =
        await moviesCollection.limit(1).get();
    expect(searchResults.size, equals(1));
    final movieFound = searchResults.docs.first.data();
    expect(movieFound.title, equals('A long time ago'));
  });
```

Reply:
Replaced

Reply:
After trying it out myself, I realized that `moviesCollection.limit` does not actually run `FakeConverterQuery.limit`. Instead it runs `MockCollectionReference.limit`. In order to force the test of `FakeConverterQuery.limit`, we have to create a query (not a collection), then convert it with a converter. Would you mind replacing it to this test:

```dart
  test('limit clips results', () async {
    final from = (snapshot, _) => Movie()..title = snapshot['title'];
    final to = (Movie movie, _) => {'title': movie.title};

    final firestore = FakeFirebaseFirestore();

    final moviesCollection = firestore
        .collection('movies')
        .withConverter(fromFirestore: from, toFirestore: to);
    await moviesCollection.add(Movie()..title = 'A long time ago');
    await moviesCollection.add(Movie()..title = 'Robot from the future');
    final rawMoviesCollection = firestore.collection('movies');
    final searchResults = await rawMoviesCollection
        .where('title', isNotEqualTo: 'Galactic')
        .withConverter(fromFirestore: from, toFirestore: to)
        .limit(1)
        .get();
    expect(searchResults.size, equals(1));
    final movieFound = searchResults.docs.first.data();
    expect(movieFound.title, equals('A long time ago'));
  });
```

I've verified on my machine that this works as intended: `rawMoviesCollection.where(...)` creates an actual MockQuery, and `MockQuery.withConverter` creates a `FakeConvertedQuery`.